Fix for translatable icon inside a multi input fields

promatik opened the pull request:
Fix for https://github.com/Laravel-Backpack/PermissionManager/issues/257.

When a Model `HasTranslations`, this fields were causing the `translatable_icon.blade.php` to fail with an error because it is expecting a single field, and in case of `checklist_dependency` and `page_or_link` fields, they have multiple. 

```
TypeError
Argument 1 passed to App\User::isTranslatableAttribute() must be of the type string, array given
```

Since it makes no sense to evaluate multiple fields, in case of `checklist_dependency` I added 2 flags, for each checklist, and in case of `page_or_link` I added the flag just in case the input `link` is translatable, which is what makes the most sense to be translated.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Laravel-Backpack/CRUD/inspections/5608aed1-c93a-4c42-ab69-7202d260c325) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Laravel-Backpack/CRUD/inspections/5608aed1-c93a-4c42-ab69-7202d260c325) completed: **No new issues**

pxpm commented:
Hello @promatik 

I'v #3276 Open. Is this the same issue?

Best, 
Pedro

promatik commented:
Hi @pxpm, sure 🤦‍♂️
I'll revert changes for `page_or_link`, since it makes more sense to check if any of the fields is translatable. Anyway, for `checklist_dependency` it makes sense to add a flag on the particular field what do you think?

promatik commented on src/resources/views/crud/fields/page_or_link.blade.php:
```diff
@@ -18,7 +18,7 @@
 @include('crud::fields.inc.wrapper_start')
     <label>{!! $field['label'] !!}</label>
-    @include('crud::fields.inc.translatable_icon')
+    @include('crud::fields.inc.translatable_icon', ['field' => ['name' => 'link']])
```
```suggestion
    @include('crud::fields.inc.translatable_icon')
```

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Laravel-Backpack/CRUD/inspections/074c8dfa-2b39-407a-97fb-162c8b5b57ed) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Laravel-Backpack/CRUD/inspections/074c8dfa-2b39-407a-97fb-162c8b5b57ed) completed: **No new issues**

pxpm commented:
Hello @promatik

I agree something need to be done, but I am not sure this will work.

If you set the `$field` to be the `primary`, in translatable icon, when we ask if `$model->isTranslatable` wouldn't we endup querying the `primary field model` instead of the current field ?

Best,
Pedro

promatik commented:
@pxpm by specifying the `field` with `['field' => $primary_dependency]`, the `translatable_icon` will run `$crud->model->isTranslatableAttribute($field['name'])` over the `$primary_dependency` field.

In this case I'm adding **two** `translatable_icon`, each with it's respective field.

I think `translatable_icon` should be ready to receive an array of names, but when we can specify which field name we want to consider to show the flag, it's better to previously do that job

pxpm commented:
@promatik I see now, thanks for clarification.

Do you think that we should do the same for `page_or_link` and add 3 flags ? Or we keep showing just one flag on `parent` field and maybe do the same here ?

promatik commented:
@pxpm, at first I though about that, but it makes no sense on `page_or_link` field.
Is it possible that someone will translate the select (`type`), and not the `link`? Or translate `page_id` and not `type` or `page_id`?

And if so, does it need to show flags for each one?
I don't think so, so one flag is ok, and the fallback of your PR  #3276 fits the condition.

tabacitu commented:
Fyi, my 2 cents, after my talk with @pxpm:
- page_or_link 1 flag (if any attribute is translatable)
- checkbox_dependency 2 flag (one for each attribute, if that attibute is translatable)

promatik commented:
Perfect @tabacitu, I agree.

> page_or_link 1 flag (if any attribute is translatable)

Fixed by #3276

> checkbox_dependency 2 flag (one for each attribute, if that attibute is translatable)

Fixed by this PR #3327

pxpm reviewed (approved):
Like we already discussed, let's do this.
